lib/amqp_client.js

```diff
@@ -219,14 +224,36 @@ AMQPClient.prototype.connect = function(url) {
   });
 };
 
-AMQPClient.prototype._resolveDeferredSenderAttaches = function(linkName, err, link) {
+/**
+ * Return a name and hash for a link associated with an address and options
+ *
+ * @param type      type of link being accessed/created
+ * @param address   the address of the link
+ * @param options   the link creation options
+ * @returns {Object}
+ */
+AMQPClient.prototype._linkInfo = function(type, address, options) {
+  var lookup = (type === 'receiver') ?
+    this._linkNameForHash.receivers : this._linkNameForHash.senders;
+
+  var hash = u.linkHash(address, options);
+  if (lookup.hasOwnProperty(hash)) {
```

Comment:
It's now possible for a client to (try and) create two links with the same name if they pass in the same options.name for differently-hashed values. I don't think that was possible before.

Adding a this._linkNamesByHash.reverseMap[name] = hash;, checking it, and throwing an error on that attempt wouldn't be hard to do, and you wouldn't need/want separate sender/receiver maps so not much extra overhead.